### PR TITLE
replace treebrowser form types

### LINKED ISSUE #48: Use the new TreeBrowserBundle form type
Use the new form type instead of the old one living in SonataDoctrinePhpcrAdminBundle.

### PR DESCRIPTION
**Pull request by ElectricMaxxx**
fix #48

early WIP. Need to really see it on dev-branch on sandbox, to find the issue.

### CODE REVIEW

**Comment by ElectricMaxxx** [comment]
I think i have no data transformation. I am really confused, why the new tree form type does not do it alone? @wouterj ping.

**Comment by wouterj** [comment]
@ElectricMaxxx take a look at the content PHPCR form type. You have to use the data transformer provided by the DoctrinePhpcrOdmBundle and configure it for the field. (this way, the tree browser bundle doesn't bind itself to a backend).

**Comment by ElectricMaxxx** [comment]
Ah that means, i have look for some kind of abstraction as this is done for almost every admin.

**Comment by ElectricMaxxx** [comment]
oh ... i think i reached some circulary issues now.

@wouterj can i use `getBuilder()` in the abstract class or do i need one created by the form builder in the admin?

**Comment by ElectricMaxxx** [comment]
i switched to the current form builder. it almost works now (miss some translations, ..)

But i don't really like that solution now. There is a complete redundance, as i don't have the document manager available in the admin directly - just the model manager. And we are completely fixed to phpcr backends only now.

**Comment by wouterj** [comment]
> But i don't really like that solution now. There is a complete redundance, as i don't have the document manager available in the admin directly - just the model manager. And we are completely fixed to phpcr backends only now.

You mean the redundance in the service config? This can be fixed using parent services: http://symfony.com/doc/current/service_container/parent_services.html

And don't worry about being fixed to PHPCR, we already are for 100%: The PHPCR ODM Admin class is extended, the `parentDocument` field only makes sense in a PHPCR context, the configure model is the PHPCR document, etc.

**Comment by ElectricMaxxx** [comment]
Ok. So we will maybe create separate ORM admins late?

Will fix the missing stuff tonight.

**Comment by ElectricMaxxx** [comment]
@wouterj shouldn't we put the the admin classes into an an phpcr context then? Means directory and namespace as we do it with the documents. Doing that after the release would be a BC-Break, i think.

File: src/DependencyInjection/CmfSonataAdminIntegrationExtension.php [hunk 1]
```diff
@@ -55,6 +55,8 @@ public function load(array $configs, ContainerBuilder $container)
         $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
 
         $this->loadBundles($config['bundles'], $loader, $container);
+
+        $loader->load('main-phpcr.xml');
```
Comment (ElectricMaxxx) [inline src/DependencyInjection/CmfSonataAdminIntegrationExtension.php]: @wouterj can you please bind that load to phpcr backend in your PR?

**Comment by wouterj** [comment]
👍 Let's merge

@kermorgant this PR should fix the issue you described yesterday. You're currently using a version of this bundle that uses an old unmaintained form type.

**Comment by dbu** [comment]
@ElectricMaxxx can you squash the commits and merge?

**Comment by kermorgant** [comment]
Wouter J writes:

>  Let's merge
>
> @kermorgant this PR should fix the issue you described yesterday. You're
> currently using a version of this bundle that uses an old unmaintained form
> type.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.*

Great news, thanks !

--
Mikael Kermorgant

**Comment by kermorgant** [comment]
Hello !

I gave the fix a try, but I'm still facing a different but somehow similar issue. The root node is now absent from the tree, and visually, the bug is the same.

I get this error message :

> No resource found at path "/menu/main/services-item" for repository "default"

Seems related to the absence of root node "cms".

https://snag.gy/ebP4YI.jpg
https://snag.gy/szoBnm.jpg

Regards
Mikael

**Comment by ElectricMaxxx** [comment]
@kermorgant thanks, can you open a separate issue please, i will have a look then. but can you have a look at your configuration and the parameters then? some of them are null now, and you have to set them explicit for the bundle. Especially the basepaths, which go directly into the tree-browser.

**Comment by wouterj** [comment]
Hmm, pretty sure this is a bug in the TreeBrowserbundle. Visually, the output seems correct (it should only display the root node defined by the form type, to avoid people to add menu's outside the menu basepath). However, the errors you get aren't expected of course.

**Comment by ElectricMaxxx** [comment]
whatching your screenshots (btw: you can direcly drag&drop the into a github comment) it looks valid, just the option to select the root as path is missing on the new treebrowser. i mentioned it in the code above.

**Comment by kermorgant** [comment]
Hello,
Thanks for the replies. I [opened a new issue here]( https://github.com/symfony-cmf/sonata-phpcr-admin-integration-bundle/issues/73).

ElectricMaxxx : Too late for the drag & drop but thanks for the tip.

**Pull request merged**